### DISCUSSION TITLE
Set up for Neovim on Apple Silicon.

### DISCUSSION THREAD

**Discussion by MaximFedarau** [community/community#19147] [Copilot Conversations] [answered]
Hello. I tried to set up Github Copilot for Neovim and I get this message:
`Copilot: 'Node.js version 16.x or 17.x required on Apple Silicon but found 18.4.
0`
Should I downgrade my Node.js, use nvm or just wait for the update?

**Comment by melvincwng** [comment]
Hi @MaximFedarau, as quoted from [GitHub Copilot Documentation](https://docs.github.com/en/copilot/getting-started-with-github-copilot/getting-started-with-github-copilot-in-neovim):
- In order to use GitHub Copilot in Neovim, "you must have Neovim and Node.js version 17 or below installed".

So unfortunately, it seems like the current workaround solution for the time being would be to downgrade the version of Node.js on your computer, should you want to use GitHub Copilot.

Hope this clarifies 😄

**Reply by melvincwng** [reply]
@MaximFedarau, if you find this reply useful, please kindly 'Mark as answer' so that others are aware of the solution, should they have the same question in the future.

Thanks and have a nice day ahead 😊

**Comment by n8henrie** [comment, marked as answer]
Because this is one of the top DDG results:

with homebrew you don't necessarily have to downgrade, you can install multiple versions simultaneously.

```console
$ brew install node@16
$ brew list node@16 | grep 'node$'
/opt/homebrew/Cellar/node@16/16.15.1/bin/node
```

Unfortunately, Copilot seems to take the first `node` it finds in the `PATH`, even if multiple are installed; I was able to run the setup with:

```console
$ PATH=/opt/homebrew/Cellar/node@16/16.15.1/bin:"${PATH}" nvim +"Copilot setup"  
```

Is there a way to set the correct path to node in configuration somewhere? If not, this would be a welcome feature!

**Reply by happyjake** [reply]
> ```
> brew list node@16 | grep 'node$'
> ```

There is an option `g:copilot_node_command` to set the node path. I set it up with packer like this.

```lua
    use {
      "github/copilot.vim",
      branch = "release",
      config = function()
        local sysname = vim.loop.os_uname().sysname
        if sysname == "Darwin" then
          vim.g.copilot_node_command = "/opt/homebrew/Cellar/node@16/16.15.1/bin/node"
        end
      end
    }

```

**Reply by MaximFedarau** [reply]
Thank you very much.

**Reply by n8henrie** [reply]
Thanks!

This minor change will give you less headaches with minor version changes:

```
vim.g.copilot_node_command = "/opt/homebrew/opt/node@16/bin/node"  
```

**Reply by DerekStride** [reply]
I didn't have any file at `/opt/homebrew/opt/node@16/bin/node` so I used the `lua-fs` module to search for the node binary in the `node@16` directory: https://github.com/DerekStride/dotfiles/commit/90ed7e5702e3786947635ec7ddaf2b33f92036aa

```lua
vim.g.copilot_node_command = vim.fs.find("node", { path = "/opt/homebrew/Cellar/node@16" })[1]
```
